internal/actor/actor.go

```
@@ -46,6 +53,22 @@ func (a *Actor) IsInternal() bool {
 	return a != nil && a.Internal
 }
 
+type userFetcher interface {
+	GetByID(context.Context, int32) (*types.User, error)
+}
+
+// User returns the expanded types.User for the actor's ID. The ID is expanded to a full
+// types.User using the fetcher, which is likely a *database.UserStore.
+func (a *Actor) User(ctx context.Context, fetcher userFetcher) (*types.User, error) {
+	a.userOnce.Do(func() {
+		a.user, a.userErr = fetcher.GetByID(ctx, a.UID)
+	})
+	if a.user.ID != a.UID {
+		panic(fmt.Sprintf("actor UID (%d) and the ID of the cached User (%d) do not match", a.UID, a.user.ID))
```

minor: you don't actually have to panic you can return an error :)